index.d.ts

```diff
@@ -1,146 +1,148 @@
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SCHEDULE_IDLE
-    scheduleIdle = 'scheduleIdle' // eslint-disable-line no-unused-vars
+    * Enable/disable specific metrics
+    */
+  collect?: {
+    cpu?: boolean,
+    eventLoopDelay?: boolean,
+    memory?: boolean,
+    gc?: boolean,
+    activeHandles?: boolean
   }
+}
 
-  enum GCOp { // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR
-    major = 'major', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_MINOR
-    minor = 'minor', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL
-    incremental = 'incremental', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB
-    weakCB = 'weakCB' // eslint-disable-line no-unused-vars
-  }
+declare interface CPUMetric {
+  /**
+    * Usage percentage
+    */
+  usage: number,
+  /**
+    * Raw value returned by `process.cpuUsage()`
+    */
+  raw: number
+}
 
-  interface GCOpStats {
-    count: number,
-    total: number,
-    average: number
-  }
+/**
+ * On Node 12 and above this is a Histogram instance from 'perf_hooks'.
+ */
+declare interface EventLoopDelayHistogram {
+  min: number,
+  max: number,
+  mean: number,
+  stddev: number,
+  percentiles: Map<number, number>,
+  exceeds: number,
+}
+declare interface EventLoopDelayMetric {
+  /**
+    * computed delay in milliseconds
+    */
+  computed: number,
+  raw: number | EventLoopDelayHistogram
+}
 
-  interface GCAggregatedEntry extends GCOpStats {
-    flags?: Map<GCFlag, GCOpStats>
-  }
+declare enum GCFlag {
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_NO
+  no = 'no', // eslint-disable-line no-unused-vars
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_CONSTRUCT_RETAINED
+  constructRetained = 'constructRetained', // eslint-disable-line no-unused-vars
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_FORCED
+  forced = 'forced', // eslint-disable-line no-unused-vars
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SYNCHRONOUS_PHANTOM_PROCESSING
+  synchronousPhantomProcessing = 'synchronousPhantomProcessing', // eslint-disable-line no-unused-vars
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_AVAILABLE_GARBAGE
+  allAvailableGarbage = 'allAvailableGarbage', // eslint-disable-line no-unused-vars
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_EXTERNAL_MEMORY
+  allExternalMemory = 'allExternalMemory', // eslint-disable-line no-unused-vars
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SCHEDULE_IDLE
+  scheduleIdle = 'scheduleIdle' // eslint-disable-line no-unused-vars
+}
+declare interface GCOpStats {
+  count: number,
+  total: number,
+  average: number
+}
+declare interface GCAggregatedEntry extends GCOpStats {
+  flags?: Map<GCFlag, GCOpStats>
+}
+declare interface GCMetric {
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR
```

Comment:
In order for TypeScript Language Servers to add in the comments as information about the properties, it needs to be in JSDoc format (ie, the `/**` style comments).

It would be better to re-write these as:

```ts
declare interface GCMetric {
  /** perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR */
  major: GCAggregatedEntry,
  /** perf_hooks.constants.NODE_PERFORMANCE_GC_MINOR */
  minor: GCAggregatedEntry,
  /** perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL */
  incremental: GCAggregatedEntry,
  /** perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB */
  weakCb: GCAggregatedEntry
}
```

(Or on multiple lines, whichever the linter prefers, etc)